From 3e8ae32c961f0a3a6ba32881e88a49d5cbf656cd Mon Sep 17 00:00:00 2001
From: Jeff Mahoney <jeffm@suse.com>
Date: Fri, 11 Jan 2019 09:57:48 -0500
Subject: README: document impact of balance + quota groups
References: fate#325529 jsc#SLE-3188

The combination of balance and quota groups as the number of references
to extents rises (either via snapshots, dedupe, or reflink copies) can
be extreme.

The file system performs quota group accounting for every extent relocated
and the result can be the appearance of stalled system as the
transaction commit can take a very long time.

Signed-off-by: Jeff Mahoney <jeffm@suse.com>
---
 README.md | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/README.md b/README.md
index 07a8104..d59f3d6 100644
--- a/README.md
+++ b/README.md
@@ -66,6 +66,10 @@ the same set of blocks is affected.
 
 The balance command uses filters to do the work in smaller batches.
 
+The impact with quota groups enabled can be extreme.  The balance operation
+performs quota group accounting for every extent being relocated, which can
+have the impact of stalling the file system for an extended period of time.
+
 __Expected result:__ If possible all the underused chunks are removed, the
 value of `total` in output of `btrfs fi df /path` should be lower than before.
 Check the logs.
-- 
2.16.4

